packages/eas-cli/src/build/ios/credentials.ts

```diff
@@ -44,6 +51,57 @@ export async function ensureIosCredentialsAsync(
   };
 }
 
+export async function setupPushKeyAsync(
```

Comment:
consider moving 
- this logic to separate Action
- calling this function from Ios credentials provider

Reply:
I moved some of this logic to a separate Action. There's some parts of this method that is specific to the build command (abort if we are building for simulator, create credentials ctx from build ctx, etc) that I left behind. Let me know if this is consistent with what you meant, or if you'd rather just get rid of the whole method and refactor it elsewhere.

Reply:
you could move
- `setupPushKeyAsync` implementation to https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/credentials/ios/IosCredentialsProvider.ts
- call `setupPushKeyAsync` in https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/credentials/ios/IosCredentialsProvider.ts#L52 after normal credentials generation

This way:
- you don't need to call `shouldProvideCredentials` because it's already verified here https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/build/ios/credentials.ts#L15
- `setupPushKeyAsync` would be simpler, because you won't need to create new context

It does not matter that much, so if you prefer to keep the current implementation I'm fine with it.